### PR TITLE
Be robust to missing content.json and schema.json

### LINKED ISSUE #26: Handle missing schema files
Make the field-preview.jsx cope with missing schema files. The current collections preview code assumes there will be one, and fails if it's missing.

### LINKED ISSUE #48: Handle missing content.json files
the tableflip-www project has a page with a content.json and a schema.json and a page with neither. No pages show up in tune. I think it's the async.map that's causing it:

https://github.com/tableflip/tune/blob/7e0ec967cd91e40d37056626c5f0ae4cb5d25502/imports/api/github/methods.js#L85

### PR DESCRIPTION
**Pull request by olizilla**
Refactors the api/github/method.js to deal with page dirs that don't have schema.json or content.json.

Next step is to hide the entry from them the user if there is no content.json to edit, but that's beyond this PR.

fixes #48
fixes #26

### CODE REVIEW

**Comment by olizilla** [comment]
The main issue I ran in to was that I couldn't see any page entries in my tune project as one page has neither a content.json or a schema.json. With this patch I can now see and edit the homepage of tableflip-www even though the startup page doesn't have those files.

**Comment by richsilv** [comment]
:eyeglasses:

File: imports/api/github/methods.js [hunk 1]
```diff
@@ -21,36 +42,27 @@ export function getLastCommit (userId, fullName, opts, cb) {
...
+        })
+      },
+      schemaJson: (cb) => {
+        getFileContents(github, fullName, dir, 'schema.json', (err, res) => {
+          cb(err, { content: res && res.content })
+        })
+      }
+    }, (err, data) => {
+      if (err) return cb(err)
+      if (!data.factsJson) return cb(err, null)
```
Comment (richsilv) [inline imports/api/github/methods.js]: super minor, but this is just `return cb()`, isn't it?

File: imports/api/github/methods.js [hunk 2]
```diff
@@ -6,6 +6,27 @@ import base64 from '/imports/lib/github/base64'
...
 
+function base64ToJson (str) {
+  var obj = {}
+  try {
+    obj = JSON.parse(base64.decode(str))
+  } catch (err) {
+    console.log(err)
+  }
+  return obj
+}
```
Comment (richsilv) [inline imports/api/github/methods.js]: :+1:

File: imports/api/github/methods.js [hunk 3]
```diff
@@ -21,36 +42,27 @@ export function getLastCommit (userId, fullName, opts, cb) {
...
-    function getRootDirContents (cb) {
-      github.getDirContents(fullName, '', cb)
-    },
-    function getFacts (res, cb) {
-      var factsJson = res.find(obj => obj.type === 'file' && obj.name === 'facts.json')
-      github.getFileContents(fullName, 'facts.json', (err, res) => {
-        cb(err, { sha: factsJson.sha, content: res.content })
+  github.getDirContents(fullName, '', (err, dir) => {
+    if (err) return cb(err) // ruh roh, couldn't list dir contents.
+    async.parallel({
```
Comment (richsilv) [inline imports/api/github/methods.js]: :+1:

**Comment by richsilv** [comment]
Very nice. :rocket:

File: imports/api/github/methods.js (hunk 1 above)
Comment (olizilla) [inline imports/api/github/methods.js]: Aye, good point.

**Pull request merged**